pkg/pillar/cmd/zedagent/zedagent.go

```diff
@@ -1232,6 +1252,50 @@ func handleZbootStatusDelete(ctxArg interface{}, key string,
 	// Nothing to do
 }
 
+func handleNodeAgentStatusModify(ctxArg interface{}, key string,
+	statusArg interface{}) {
+
+	getconfigCtx := ctxArg.(*getconfigContext)
+	status := cast.NodeAgentStatus(statusArg)
+	if status.Key() != key {
+		log.Errorf("handleNodeAgentStatusModify: invalid key %s\n", key)
+		return
+	}
+	updateInprogress := getconfigCtx.updateInprogress
+	ctx := getconfigCtx.zedagentCtx
+	ctx.remainingTestTime = status.RemainingTestTime
+	getconfigCtx.updateInprogress = status.UpdateInprogress
+	if status.NeedsReboot {
+		initiateDeviceReboot(ctx, status.RebootReason)
+	} else {
+		// if config reboot command was initiated and
```

Comment:
There is also a case of an update which is deferred due to the previous update still being tested. Where do you handle that?

Reply:
We handle this case, in this function itself.